### PR TITLE
Fix/retrieval api not returning all responses

### PR DESCRIPTION
**Pull request by craigzour**
# Summary | Résumé

- Fixed major bug in the Retrieval API which would not return all existing responses. This is due to some QueryCommand SDK [limitations](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html) not being fully taken into account.

> A single Query operation will read up to the maximum number of items set (if using the Limit parameter) or a maximum of 1 MB of data and then apply any filtering to the results using FilterExpression. If LastEvaluatedKey is present in the response, you will need to paginate the result set. For more information, see [Paginating the Results](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html#Query.Pagination) in the Amazon DynamoDB Developer Guide.

- Introduced new [library](https://aws.amazon.com/blogs/developer/mocking-modular-aws-sdk-for-javascript-v3-in-unit-tests/) which allows us to mock AWS SDK operations.
- Refactored Retrieval API unit tests to use the new library
- Added two tests against new Retrieval API implementation

# Test instructions | Instructions pour tester la modification

- Using the old and then the new code, try to retrieve responses from form ID 19 or 309. You should see more responses returned by the new code.

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [x] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: pages/api/id/[form]/retrieval.ts [hunk 1]
```diff
@@ -49,48 +54,66 @@ function connectToDynamo(): DynamoDBDocumentClient {
...
+        FilterExpression: "FormID = :formID",
+        //A filter expression cannot contain partition key or sort key attributes.
+        //You need to specify those attributes in the key condition expression, not the filter expression.
+        ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved,SecurityAttribute",
+      };
+
+      const response = await documentClient.send(new QueryCommand(getItemsDbParams));
+
+      if (response.Items) {
+        for (const item of response.Items) accumulatedResponses.push(item);
```
Comment (Moro-Code) [inline pages/api/id/[form]/retrieval.ts:99]: Instead of doing this you could use the concat function 

`accumulatedResponses.concat(response.Items)`

**Review by Moro-Code — COMMENTED** [review]
Just had a comment about using a loop when an array method can be used

File: pages/api/id/[form]/retrieval.ts (hunk 1 above)
Comment (craigzour) [reply pages/api/id/[form]/retrieval.ts:99]: Absolutely! No idea why I did not do that initially  😕

**Review by Moro-Code — APPROVED** [review]
Looks good lets merge it ! Thanks for the work @craigzour

**Pull request merged**